src/flake8/options/config.py

```diff
@@ -120,7 +124,20 @@ def parse_config(
         LOG.debug('Option "%s" returned value: %r', option_name, value)
 
         final_value = option.normalize(value, cfg_dir)
+
+        if option_name in {"ignore", "extend-ignore"}:
+            for error_code in final_value:
+                if not VALID_CODE.match(error_code):
+                    raise configparser.ParsingError(
+                        f"Error code {error_code} "
+                        f"supplied to '{option_name}' option "
+                        f"does not match {VALID_CODE.pattern}"
+                    )
+
         assert option.config_name is not None
         config_dict[option.config_name] = final_value
 
     return config_dict
+
+
+VALID_CODE = re.compile("^[A-Z]{1,3}[0-9]{0,3}$", re.ASCII)
```

Comment:
you've got this defined in a couple places (3?) -- probably want to bring them all together

Reply:
Good idea! I imported the constant from finder.py now -- I considered moving it upwards into utils.py but decided against it with the contribution guidelines in mind (no extraneous refactoring), but you could make a case that it's better to import downwards (both import from utils.py) than sideways as we do now. What are your thoughts on this?